sdk/trace/src/main/java/io/opentelemetry/sdk/trace/export/BatchSpanProcessor.java

```diff
@@ -131,7 +132,7 @@ public CompletableResultCode forceFlush() {
 
     private long nextExportTime;
     private final BlockingQueue<ReadableSpan> queue;
-    private final AtomicBoolean needSignal = new AtomicBoolean(false);
+    private final AtomicLong spansNeeded = new AtomicLong(Long.MAX_VALUE);
```

Comment:
should this be initialized to `maxExportBatchSize`?

Reply:
I am using MAX_VALUE to imply that exporter thread is not expecting any signal. When it needs a signal, it is updating the value to number of spans it requires before doing the IO. Since in the beginning exporter thread is not expecting any signal, setting it to MAX_VALUE seems reasonable.

Reply:
cool. can you add a comment in the code about that, so when we look at this in 3 months, we don't have to dig through PRs to figure out why?  :)